ansible/playbooks/roles/opensearch/tasks/set_facts.yml

```diff
@@ -0,0 +1,12 @@
+---
+- name: Set the name of binary to download for x86_64 family
```

Comment:
How about adding to roles/opensearch/defaults/main.yml:
```
file_names:
  opensearch:
    x86_64: opensearch-1.2.4-linux-x64.tar.gz
    aarch64: opensearch-1.2.4-linux-arm64.tar.gz
  # Perftop is not supported on ARM (https://github.com/opensearch-project/perftop/issues/26)
  opensearch_perftop: opensearch-perf-top-1.1.0.0-linux-x64.zip
```
Then when condition would be not needed since you could get filename like this:
`file_names.opensearch[ansible_architecture]`

Moreover, it looks like we may need to use v1.2.0.0 instead of 1.1.0.0 since they wrote on GitHub:
> Version 1.2.0.0 (Version compatible with OpenSearch 1.2.0)
> Version 1.1.0.0 (Version compatible with OpenSearch 1.1.0)

Reply:
UR right. I've taken regard to this and implemented logic to distinct the installation of PerTop based on the arch .

Reply:
Moved to defaults/main now.